src/Features/CSharp/Portable/Completion/CompletionProviders/DeclarationNameCompletionProvider.cs

```diff
@@ -0,0 +1,184 @@
+                completionContext.AddItem(CreateCompletionItem(name, GetGlyph(kind, nameInfo.DeclaredAccessibility), sortValue++.ToString("D8")));
+            }
+
+            AddBuilder(completionContext);
+        }
+
+        private bool IsValidType(ITypeSymbol type)
+        {
+            if (type == null)
+            {
+                return false;
+            }
+
+            if (type.IsErrorType() && (type.Name == "var" || type.Name == string.Empty))
+            {
+                return false;
+            }
+
+            return type.SpecialType != SpecialType.System_Void;
+        }
+
+        private Glyph GetGlyph(SymbolKind kind, Accessibility declaredAccessibility)
+        {
+            Glyph publicIcon;
+            switch (kind)
+            {
+                case SymbolKind.Field:
+                    publicIcon = Glyph.FieldPublic;
+                    break;
+                case SymbolKind.Local:
+                    publicIcon = Glyph.Local;
+                    break;
+                case SymbolKind.Method:
+                    publicIcon = Glyph.MethodPublic;
+                    break;
+                case SymbolKind.Parameter:
+                    publicIcon = Glyph.Parameter;
+                    break;
+                case SymbolKind.Property:
+                    publicIcon = Glyph.PropertyPublic;
+                    break;
+                case SymbolKind.RangeVariable:
+                    publicIcon = Glyph.RangeVariable;
+                    break;
+                case SymbolKind.TypeParameter:
+                    publicIcon = Glyph.TypeParameter;
+                    break;
+                default:
+                    throw new ArgumentException();
+            }
+
+            switch (declaredAccessibility)
+            {
+                case Accessibility.Private:
+                    publicIcon += Glyph.ClassPrivate - Glyph.ClassPublic;
+                    break;
+
+                case Accessibility.Protected:
+                case Accessibility.ProtectedAndInternal:
+                case Accessibility.ProtectedOrInternal:
+                    publicIcon += Glyph.ClassProtected - Glyph.ClassPublic;
+                    break;
+
+                case Accessibility.Internal:
+                    publicIcon += Glyph.ClassInternal - Glyph.ClassPublic;
+                    break;
+            }
+
+            return publicIcon;
+        }
+
+        private void AddBuilder(CompletionContext completionContext)
+        {
+            completionContext.SuggestionModeItem = CommonCompletionItem.Create(CSharpFeaturesResources.Name, CompletionItemRules.Default);
+        }
+
+        private ITypeSymbol UnwrapType(ITypeSymbol type)
+        {
+            var nts = type as INamedTypeSymbol;
+            if (nts != null && nts.ConstructedFrom != null)
+            {
+                var constructedFrom = nts.ConstructedFrom;
```

Comment:
just use .OriginalDefinition.  ConstructedFrom is really for extension method cases.